go/README.md

```diff
@@ -0,0 +1,110 @@
+
+## Arrow memory management
+Understanding this is the trickiest part of this integration.
+
+At a high level, when using the Python<>Go integration, the python layer exports request data into an [Arrow Record batch](https://arrow.apache.org/docs/python/data.html) which is transferred to go using Arrow's zero copy mechanism.
+Similarly, the go layer converts feature values read from the online store into a Record Batch that's exported to Python using the same mechanics.
+
+The first thing to note is that from the python perspective, all the export logic assumes that we're exporting to & importing from C, not Go. This is because pyarrow only interops with C, and the fact we're using Go is an implementation detail not relevant to the python layer.
+
+### Export Entities & Request data from Python to Go
+The code exporting to C is this, in [online_feature_service.py](../sdk/python/feast/embedded_go/online_features_service.py)
+```
+(
+    entities_c_schema,
+    entities_ptr_schema,
+    entities_c_array,
+    entities_ptr_array,
+) = allocate_schema_and_array()
+(
+    req_data_c_schema,
+    req_data_ptr_schema,
+    req_data_c_array,
+    req_data_ptr_array,
+) = allocate_schema_and_array()
+
+batch, schema = map_to_record_batch(entities, join_keys_types)
+schema._export_to_c(entities_ptr_schema)
+batch._export_to_c(entities_ptr_array)
+
+batch, schema = map_to_record_batch(request_data)
+schema._export_to_c(req_data_ptr_schema)
+batch._export_to_c(req_data_ptr_array)
+```
+
+Under the hood, `allocate_schema_and_array` allocates a pointer (`struct ArrowSchema*` and `struct ArrowArray*`) in native memory (i.e. the C layer) using cffi.
+Next, the RecordBatch us exporting to this pointer using [`_export_to_c`](https://github.com/apache/arrow/blob/master/python/pyarrow/table.pxi#L2509), which uses [`ExportRecordBatch`](https://arrow.apache.org/docs/cpp/api/c_abi.html#_CPPv417ExportRecordBatchRK11RecordBatchP10ArrowArrayP11ArrowSchema) under the hood.
+
+As per the documentation for ExportRecordBatch:
+> Status ExportRecordBatch(const RecordBatch &batch, struct ArrowArray *out, struct ArrowSchema *out_schema = NULLPTR)
+> Export C++ RecordBatch using the C data interface format.
+> 
+> The record batch is exported as if it were a struct array. The resulting ArrowArray struct keeps the record batch data and buffers alive until its release callback is called by the consumer.
+
+This is why `GetOnlineFeatures()` in `online_features.go` calls `record.Release()` the following:
+```
+resp, err := s.fs.GetOnlineFeatures(
+    context.Background(),
+    featureRefs,
+    featureService,
+    entitiesProto,
+    requestDataProto,
+    fullFeatureNames)
+
+entitiesRecord.Release()
+requestDataRecords.Release()
+
+```
+
+Additionally, we need to pass in a pair of pointers into which are set in `GetOnlineFeatures()` so that the resultant feature values can be passed back to Python (via the C layer) using zero-copy semantics.
```

Comment:
maybe rephrase to, "Additionally, we need to pass in a pair of pointers to `GetOnlineFeatures()` that are populated by Go and the resultant populated feature values pointers(in the C layer) are passed back to Python via zero-copy semantics.